product-scenarios/1-manage-public-partner-private-apis/1.4-delete-an-existing-api/1.4.1-delete-an-existing-api/src/test/java/org/wso2/am/scenario/tests/delete/existing/api/DeleteAPINegativeTestCases.java

```diff
@@ -0,0 +1,228 @@
+            changeApiState(name, state);
+        }
+        checkDeleteAPI(name, "Cannot remove the API as active ");
+        verifyAPIAvailableInPublisher(name);
+//        check availability in store for only PUBLISHED apis
+        if (state.equals(APILifeCycleState.PUBLISHED)) {
+            verifyAPIAvailableInStore(name);
+        }
+    }
+
+    @Test(description = "1.4.1.5")
+    public void testDeleteAPIByUnauthorizedUser() throws Exception {
+        String name = API_NAME_PREFIX + "unauthUser";
+        createApi(name);
+        createUserWithPublisherRole(API_PUBLISHER_USERNAME, API_PUBLISHER_PW, ADMIN_USERNAME, ADMIN_PW);
+        apiPublisher.login(API_PUBLISHER_USERNAME, API_PUBLISHER_PW);
+        checkDeleteAPI(name,
+                "does not have the required permission: /permission/admin/manage/api/create");
+        apiPublisher.login(API_CREATOR_PUBLISHER_USERNAME, API_CREATOR_PUBLISHER_PW);
+//        API is in CREATED state therefor only check availability in publisher
+        verifyAPIAvailableInPublisher(name);
+    }
+
+    @Test(description = "1.4.1.6")
+    public void testDeleteNonExistingAPI() throws Exception {
+        checkDeleteAPI(API_NAME_PREFIX + "nonExist", "Unable to find the API");
+    }
+
+    private void createAPIAndSubscribe(String apiName, String applicationName) throws Exception{
+        createApplication(applicationName);
+        createApi(apiName);
+        changeApiState(apiName, APILifeCycleState.PUBLISHED);
+        subscribeToAPI(apiName, applicationName);
+    }
+
+    private void createApplication(String applicationName) throws Exception{
+        HttpResponse addApplicationResponse = apiStore
+                .addApplication(applicationName,
+                        APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, "", "");
+        applicationsList.add(applicationName);
+        verifyResponse(addApplicationResponse);
+        assertEquals(new JSONObject(addApplicationResponse.getData()).get("status"), "APPROVED",
+                "Application creation failed for application: " + applicationName);
+    }
+
+    private void createApi(String apiName) throws Exception {
+        APIRequest apiRequest = new APIRequest(apiName, "/" + apiName, "public", API_VERSION,
+                "/menu", APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED,
+                new URL("https://localhost:9443/am/sample/pizzashack/v1/api/"));
+        HttpResponse createAPIResponse = apiPublisher.addAPI(apiRequest);
+        apiList.add(apiName);
+        verifyResponse(createAPIResponse);
+        HttpResponse apiInfo = apiPublisher.getAPI(apiName, API_CREATOR_PUBLISHER_USERNAME, API_VERSION);
+        verifyResponse(apiInfo);
+    }
+
+    private void changeApiState(String apiName, APILifeCycleState state) throws Exception {
+        org.wso2.am.integration.test.utils.bean.APILifeCycleStateRequest updateRequest;
```

Comment:
Why don't define the class import statement at the top of the class?

Reply:
Fixed with #5210